src/main/java/seedu/address/storage/JsonAdaptedGroup.java

```diff
@@ -0,0 +1,63 @@
+package seedu.address.storage;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.group.Group;
+import seedu.address.model.person.Person;
+
+/**
+ * Jackson-friendly version of {@link Group}.
+ */
+class JsonAdaptedGroup {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Group's %s field is missing!";
+
+    private final String groupName;
+    private final List<JsonAdaptedPerson> members = new ArrayList<>();
+
+    /**
+     * Constructs a {@code JsonAdaptedGroup} with the given group details.
+     */
+    @JsonCreator
+    public JsonAdaptedGroup(@JsonProperty("groupName") String groupName,
+                            @JsonProperty("members") List<JsonAdaptedPerson> members) {
+        this.groupName = groupName;
+        if (members != null) {
+            this.members.addAll(members);
+        }
+    }
+
+    /**
+     * Converts a given {@code Group} into this class for Jackson use.
+     */
+    public JsonAdaptedGroup(Group source) {
+        groupName = source.getGroupName().toString();
+        members.addAll(source.getMembers().stream()
+                .map(JsonAdaptedPerson::new)
+                .collect(Collectors.toList()));
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted group object into the model's
+     * {@code Group} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in
+     *                               the adapted group.
+     */
+    public Group toModelType() throws IllegalValueException {
```

Comment:
Just need add a test for the saving process next week and we good on code coverage!!